pages/api/enroll/child.ts

```diff
@@ -174,6 +177,48 @@ export default async function handle(
                 return;
             }
 
+            // check if there exists waitlist records for this class
+            const waitlistRecords = await getWaitlistRecordsByClassId(
+                parentRegistrationInput.classId,
+            );
+            // if there are, get class information & send email to all on waitlist
+            if (waitlistRecords.length > 0) {
+                const waitlistClass = await getClass(
+                    parentRegistrationInput.classId,
+                );
+                if (!waitlistClass) {
+                    ResponseUtil.returnBadRequest(
+                        res,
+                        `Class information could not be found`,
+                    );
+                    return;
+                }
+
+                const mailerPromises = [];
+                waitlistRecords.forEach((record) => {
+                    const emailSubject = `Spot Available for Waitlisted Class: ${waitlistClass.name}`;
+                    mailerPromises.push(
+                        send(
+                            process.env.EMAIL_FROM,
+                            (record as any).parent.user.email,
```

Comment:
oh yeah also a point about this ... i was getting an error saying "property does not exist on value of type" and was reading about how to fix ([stackoverflow](https://stackoverflow.com/questions/18083389/ignore-typescript-errors-property-does-not-exist-on-value-of-type)) --

if i leave it as `record.parent.user.email` i just get the red squiggly line under `parent` but everything executes fine; i'll take a closer look at this error later + get it fixed before merge !

Reply:
yea this is an instance of prisma not playing well with typescript. You can get away with this by telling typescript to trust Prisma return types. 

```typescript
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
async function getWaitlistRecordsByClassId(classId: number) {
```

then, the type returned will have `.parent` as you expect.